### PR TITLE
Source SalesForce: switch to REST if BulkNotSupportedException

### PR DESCRIPTION
**Pull request by artem1205**
## What

Resolving https://github.com/airbytehq/airbyte-internal-issues/issues/9809

Switch to Rest API if bulk job failed 

## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-integrations/connectors/source-salesforce/source_salesforce/rate_limiting.py [hunk 1]
```diff
@@ -126,6 +128,11 @@ def interpret_response(self, response: Optional[Union[requests.Response, Excepti
             f"An error occurred: {response.content.decode()}",
         )
 
+    @staticmethod
+    def _is_bulk_job_status_check(response: requests.Response) -> bool:
+        # TODO comment on PR: I don't like that because it duplicates the format of the URL but with a test at least we should be fine to valide once it changes
+        return bool(re.compile(r"/services/data/v\d{2}\.\d/jobs/query/[^/]+$").search(response.url))
```
Comment (maxi297) [inline airbyte-integrations/connectors/source-salesforce/source_salesforce/rate_limiting.py:134]: Abort and delete both have the same URL but their methods are different. Should we match on the HTTP method POST as well?

File: airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py [hunk 2]
```diff
@@ -336,6 +336,8 @@ def _read_pages(
         # Always return an empty generator just in case no records were ever yielded
         yield from []
 
+    _read_single_page = _read_pages
```
Comment (maxi297) [inline airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py:339]: This solution should not be needed after https://github.com/airbytehq/airbyte/pull/45601 is merged. I'll work on that today

**Review by maxi297 — COMMENTED** [review]
Just a couple of early comments. Poke me when this is ready for review

File: airbyte-integrations/connectors/source-salesforce/source_salesforce/rate_limiting.py [hunk 3]
```diff
@@ -126,6 +128,11 @@ def interpret_response(self, response: Optional[Union[requests.Response, Excepti
             f"An error occurred: {response.content.decode()}",
         )
 
+    @staticmethod
+    def _is_bulk_job_status_check(response: requests.Response) -> bool:
+        # TODO comment on PR: I don't like that because it duplicates the format of the URL but with a test at least we should be fine to valide once it changes
```
[the remark this answers is not in the document]
Comment (artem1205) [reply airbyte-integrations/connectors/source-salesforce/source_salesforce/rate_limiting.py]: refactored

File: airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py (hunk 2 above)
Comment (artem1205) [reply airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py:339]: Got it, I'll remove this line afterward​s.

Comment (maxi297) [reply airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py:339]: The fix is [here](https://github.com/airbytehq/airbyte/pull/45626/files#diff-0572edeedd561901f8110c0f88b48efde87d1ed91247c0eb189c7229da12090dR113-R123)

**Review by maxi297 — APPROVED** [review]
LGTM! Can we confirm that `test_given_bulk_restrictions_when_read_then_switch_to_standard` and `test_given_job_is_failed_when_read_then_switch_to_standard` are now passing? I'll approve conditional to that

**Pull request merged**